scripts/symbolize.py

```diff
@@ -0,0 +1,177 @@
+# and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+"""Symbolizes OP-TEE abort dumps
+
+This scripts reads an OP-TEE abort message from stdin and adds debug
+information ('function at file:line') next to each address in the call stack.
+It uses the paths provided on the command line to locate the appropriate ELF
+binary (tee.elf or Trusted Application) and runs arm-linux-gnueabihf-addr2line
+or aarch64-linux-gnu-addr2line to process the addresses.
+
+Inspired by a script of the same name by the Chromium project.
+
+Sample usage:
+
+$ scripts/symbolize.py -d out/arm-plat-hikey/core -d ../optee_test/out/ta/*
+<paste dump here>
+^D
+
+"""
+
+import argparse
+import glob
+import re
+import subprocess
+import sys
+
+TA_UUID_RE = re.compile(r'Status of TA (?P<uuid>[0-9a-f\-]+)')
+TA_INFO_RE = re.compile(':  arch: (?P<arch>\w+)  '
+                        'load address: (?P<load_addr>0x[0-9a-f]+)');
+CALL_STACK_RE = re.compile('Call stack:')
+STACK_ADDR_RE = re.compile(r':  (?P<addr>0x[0-9a-f]+)')
+X64_REGS_RE = re.compile(':  x0  [0-9a-f]{16} x1  [0-9a-f]{16}');
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--dir', action='append', nargs='+',
+        help='Search for ELF file in DIR. tee.elf is needed to decode '
+             'a TEE Core or pseudo-TA abort, while <TA_uuid>.elf is required '
+             'if a user-mode TA has crashed.')
+    parser.add_argument('-s', '--strip_path',
+        help='Strip prefix from file paths')
+
+    return parser.parse_args()
+
+class Symbolizer(object):
+    def __init__(self, out, dirs, strip_path):
+        self._out = out
+        self._dirs = dirs
+        self._strip_path = strip_path
+        self._addr2line = None
+        self._bin = 'tee.elf'
+        self.reset()
+
+    def get_elf(self, elf_or_uuid):
+        if not elf_or_uuid.endswith('.elf'):
+            elf_or_uuid += '.elf'
+        for d in self._dirs:
+            elf = glob.glob(d + '/' + elf_or_uuid)
+            if elf:
+                return elf[0]
+
+    def spawn_addr2line(self):
+        if not self._addr2line:
+	    elf = self.get_elf(self._bin);
```

Comment:
From this line to 101 you are using leading tabs, which makes Python unhappy (you are using leading spaces in the rest of the script).